pkg/controller/integration/kits.go

```diff
@@ -82,6 +82,18 @@ func lookupKitsForIntegration(ctx context.Context, c ctrl.Reader, integration *v
 	return kits, nil
 }
 
+// sameOrMatch returns whether the v1.IntegrationKit ios the one used by the v1.Integration or if it meets the
+// requirements of the v1.Integration.
+func sameOrMatch(kit *v1.IntegrationKit, integration *v1.Integration) (bool, error) {
+	if integration.Status.IntegrationKit != nil {
+		if integration.Status.IntegrationKit.Namespace == kit.Namespace && integration.Status.IntegrationKit.Name == kit.Name {
+			return true, nil
+		}
+	}
+
+	return integrationMatches(integration, kit)
```

Comment:
I've just fixed another possibly related issue here #3487. In that case it was failing because the `match` always returned false when an IntegrationKit was on error. Not sure if there is any check in the `integrationMatches` that is influencing the failure here, and if it makes sense to revisit it. I'd expect the `integrationMatches` to take care of those new added conditions as well.

Reply:
For what I understood, in the case the status of a kit should not influence the trigger as in fact every change to a kit should result in a reconcile being triggered. Then the integration controller should decide what to do so I guess your changes are improving also this case.